ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -1020,6 +1020,18 @@ def reboot_dut(self):
         if stderr != []:
             self.log("stderr from %s: %s" % (self.reboot_type, str(stderr)))
         self.log("return code from %s: %s" % (self.reboot_type, str(return_code)))
+        # Check sonic version after reboot
+        target_version = self.test_params['target_version']
+        if target_version:
+            stdout, stderr, return_code = self.dut_connection.execCommand("sudo sonic_installer list | grep Current | awk '{print $2}'")
+            current_version = ""
+            if stdout != []:
+                current_version = str(stdout[0]).replace('\n', '')
+            self.log("Current={} Target={}".format(current_version, target_version))
+            if current_version != target_version:
+                self.fails['dut'].add("Sonic upgrade failed. Target={} Current={}".format(\
+                    target_version, current_version))
+                thread.interrupt_main()
```

Comment:
Please define a function for this chunk of code handling image installation. You need to remove /host/old_config/config_db.json if /hsot/old_config/minigraph.xml exists. (See reference here: https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/reduce_and_add_sonic_images.py#L80). Or you can invoke this library directly.

Reply:
The code is refactored. Now reduce_and_add_sonic_images.py is reused. Thanks.